mypy/applytype.py

```diff
@@ -62,3 +68,32 @@ def apply_generic_arguments(callable: CallableType, types: List[Type],
         ret_type=expand_type(callable.ret_type, id_to_type),
         variables=remaining_tvars,
     )
+
+
+def get_incompatible_arg_constraints(arg_types: Sequence[Type], type: Type,
+                                     index: int) -> Dict[str, Tuple[str]]:
+    """Gets incompatible function arguments with the constrained types.
+
+    An example of a constrained type is AnyStr which must be all str or all byte.
+    """
+    constraints = {}  # type: Dict[str, Tuple[str]]
+    if isinstance(type, Instance) and type.type.name() == 'object':
```

Comment:
Is there another simple check to see if the constraints are being violated? I write this as a pretty restrictive check which means the rest of the code won't be executed very often (I was worried about adding too much overhead which might be pre-mature optimization). Maybe see if `type.type.name() != arg_types[index].values`? But it seems like there must be a better solution.